crates/extensions/c8y_mapper_ext/src/actor.rs

```diff
@@ -148,9 +150,16 @@ impl C8yMapperActor {
         // If incoming message follows MQTT topic scheme v1
         if let Ok((_, channel)) = self.converter.mqtt_schema.entity_channel_of(&message.topic) {
             match self.converter.try_register_source_entities(&message).await {
-                Ok(pending_entities) => {
-                    self.process_registered_entities(pending_entities).await?;
-                }
+                Ok((outcome, pending_entities)) => match outcome {
+                    UpdateOutcome::Inserted => {
+                        self.process_registered_entities(pending_entities).await?
+                    }
+                    UpdateOutcome::Updated(updated_entity, old_entity) => {
+                        self.process_entity_update(updated_entity, old_entity)
+                            .await?
+                    }
+                    UpdateOutcome::Deleted | UpdateOutcome::Unchanged => (),
+                },
```

Comment:
This match could be used a guide to refactor the methods returning `(UpdateOutcome, Vec<RegisteredEntityData>`. The pending entities seems to be only used when the outcome is `Inserted` and even populated only in that case.

Something as bellow seems more appropriate:

```rust
pub enum UpdateOutcome {
    Unchanged,
    Inserted(Vec<RegisteredEntityData>),
    Updated(EntityMetadata, EntityMetadata),
    Deleted,
}
```

Reply:
Yeah, I originally wanted to do it exactly as you proposed, with the `Inserted` variant holding the `Vec<RegisteredEntityData>`. But did fall back to this, as I felt that refactoring would be a bit costly and complicate this PR. But now that the other changes have already complicated it, may be its worth doing that refactoring as well. Will give it a shot.

Reply:
Resolved by a81147de6